### DISCUSSION TITLE
IVL_UVM now uploaded

### DISCUSSION THREAD

**Discussion by svenka3** [steveicarus/iverilog#430] [General]
Please see:

https://github.com/svenka3/ivl_uvm

It is a heavy WIP project with limited bandwidth being spent on a voluntary basis. I would like to see the basic tests (We have 3 as of now) to be part of ivltests so that we keep the code sane and not break from time-to-time.

I am looking for ideas on how to make the tests into a regressible format if there are suggestions. 

Thanks in advance for any comments.

Cheers
Srini

**Comment by caryr** [comment]
Thanks, I will look at this soon.

**Comment by svenka3** [comment]
Quick update - I now have a tiny APB VIP (Just driver) code running with IVL_UVM! This means a lot to me personally as I believe we now have:

1. Simple UVM like messaging
2. Verbosity control working
3. Few common CLP (Command Line Processor) options now working - TESTNAME, TIMEOUT
4. Basic Mailbox functionality
5. Basic TLM - pull port style working

And then using the above (all in class based), we have driver, top, interface - all in module - using the above infrastructure.

Code clean-up pending, but I have pushed it to IVL_UVM repo as a backup anyway.

Cheers
